research/conditional/utils/model_utils.py

```diff
@@ -219,14 +219,26 @@ def get_attention_layer(args):
     return attention_layer_fun
 
 
-def get_residual_layer(args):
-    if args.norm_class == "layer_norm":
-        norm_class = LayerNorm
-    elif args.norm_class == "rms_norm":
-        norm_class = llm.RMSNorm
+def get_norm_class(norm_class):
+    if norm_class == "layer_norm":
+        return LayerNorm
+    elif norm_class == "rms_norm":
+        return llm.RMSNorm
     else:
-        raise NotImplementedError(f"Norm type {args.norm_class} not implemented")
+        raise NotImplementedError(f"Norm type {norm_class} not implemented")
+
+
+def get_residual_layer(args):
+    norm_class = get_norm_class(args.norm_class)
     if args.residual_mode == "pre_norm":
+        if args.parallel_blocks:
+            return partial(
+                llm.PreNormBlock, dmodel=args.dmodel, norm_class=torch.nn.Identity
```

Comment:
Can we make e.g. `llm.ParallelPreNormBlock`? Now it looks like we have some part of the logic for creating a block in [`research/conditional/utils/model_utils.py`, `research/conditional/train/cc_train.py`] and another part in `lizrd/core/llm.py`, so basically it's split between `research` and `lizrd`, while actually it should reside in `lizrd`.